src/lib/Gen/hbs/HandlebarsGenerator.cpp

```diff
@@ -15,13 +15,73 @@
 #include "Builder.hpp"
 #include "MultiPageVisitor.hpp"
 #include "SinglePageVisitor.hpp"
+
+#include "lib/Lib/TagfileWriter.hpp"
+#include "lib/Support/RawOstream.hpp"
+
+#include <llvm/ADT/SmallString.h>
+#include <llvm/Support/FileSystem.h>
+#include <llvm/Support/Path.h>
+
 #include <mrdocs/Support/Path.hpp>
+
+#include <fstream>
 #include <sstream>
 
 namespace clang {
 namespace mrdocs {
 namespace hbs {
 
+namespace {
+
+/** Generate a Tagfile for cross-referencing. */
+Error
+generateTagfile(
+    HandlebarsCorpus const& hbsCorpus,
+    std::string_view tagFilePath,
+    std::string_view generatedFilesPath,
+    bool isSinglePage)
+{
+    if (tagFilePath.empty())
+        return Error::success();
```

Comment:
That's not a success :)

The caller should ensure the tagfile is not empty, so it's a precondition: `MRDOCS_ASSERT(!tagFilePath.empty());`